### ISSUE TITLE
AutoSuggestBox inside NavigationView truncates text when text scaling applied

### ISSUE THREAD

**Issue by Jamatth** [microsoft/microsoft-ui-xaml#6678]
### Describe the bug

When using an AutoSuggestBox inside a NavigationView (using the NavigationView.AutoSuggestBox property) and you increase the text size through the settings app, the AutoSuggestBox stays the same height and the text gets truncated at the bottom.

The AutoSuggestBox when used on its own _does_ increase in height when you apply text scaling, so that all the text fits in the box.

The behavior of the AutoSuggestBox inside the NavigationView should be fixed so that when text scaling is applied, the box expands.  This will help low visibility users not get confused by truncated text.

### Steps to reproduce the bug

1. Create a blank UWP
2. Add a NavigationView with an AutoSuggestBox
3. Launch the app and type some text in the box
4. Go to the Settings app > Accessibility > Text size and increase to 225%
5. Observe that the AutoSuggestBox does not grow enough in height and the text in the box gets truncated

### Expected behavior

When using just an AutoSuggestBox (not inside a NavigationView) increasing the text size also increases the height of the AutoSuggestBox so that the full text fits in the box.  The AutoSuggestBox should also do this when placed inside a NavigationView

### Screenshots

This shows an AutoSuggestBox inside a NavigationView truncating the bottom of the text, while an AutoSuggestBox by itself does not truncate the bottom.
![image](https://user-images.githubusercontent.com/93150350/152569269-e072c19e-5e57-47cb-8905-afc6b3a4109b.png)

### Windows app type

- [X] UWP
- [ ] Win32

### Device form factor

Desktop

**Comment by StephenLPeters** [comment]
NavigationView's template should allow for the Autosuggest box to expand vertically in this scenario.

**Comment by StephenLPeters** [comment]
@karenbtlai FYI
